docs/ASM.md

```diff
@@ -84,8 +84,8 @@ https://github.com/chef/knife-windows#nodes
                                         be unique within the deployment.
       :azure_os_disk_name               Optional. Specifies the friendly name of the disk containing
                                         the guest OS image in the image repository.
-      :azure_vm_size                    Size of virtual machine. Default is Small.
-                                        (ExtraSmall, Small, Medium, Large, ExtraLarge)
+      :azure_vm_size                    Size of virtual machine. Default is Small. Accepts all valid size of virual machine.
```

Comment:
Aren't these values hardcoded in `lib/azure/resource_management/ARM_base.rb` on line 23? Isn't there a risk these will change and then this would be incorrect about accepting all valid sizes?

Not a big deal, but I would drop that  part of the line and just keep "Size of virtual machine. Default is Small"

Either way, there's a typo in "virual"

Reply:
@btm Yes you are right about if the hard coded sizes change in future then that would be incorrect behaviour. What we can do is just keep default as Standard_A1 instead of small and remove all other hard coded values or not to set any default one and make this option as mandatory. Let me know your thoughts on this and we will change the code accordingly.

Reply:
I'd just keep one as default, yeah. Maybe put a couple in the README.md as examples of excatly what they would look like with a link to a web page like https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes that lists all the sizes.

Then I'd test what happens if we send an invalid value to Azure and see what error we get. If it's not a particularly helpful error, I'd look for how to catch it and return an error like `"Invalid VM size: #{azure_vm_size}"`.

Reply:
@btm, @Vasu1105 when we put invalid VM size in `knife azurerm server create` command, then we will get error like this:
```
The requested VM size Stanvdardf_F2 is not available in the current region. The sizes available in the current region are: Standard_A0,Standard_A1,Standard_A2,Standard_A3,Standard_A5,Standard_A4,Standard_A6,Standard_A7,Basic_A0,Basic_A1,Basic_A2,Basic_A3,Basic_A4,Standard_D1,Standard_D2,Standard_D3,Standard_D4 .......
```
This error is pretty much clear for the users, so I think we don't need to catch this error. Please let me know your thoughts on this.

Reply:
@kapilchouhan99 yeah that feels fine.